### DISCUSSION TITLE
Can the Operator deploy and manage multiple OpenSearch clusters?

### DISCUSSION THREAD

**Discussion by mcaliandro** [Opster/opensearch-k8s-operator#661] [General]
Hi all!
I would like to use one deployed instance of the Operator in a Kubernetes cluster to spawn multiple OpenSearch clusters in different namespaces.
Before doing it by myself in a testing environment, I would like to know if somebody has already tried and, eventually, succeeded to deploy it.
Maybe, it would be worthy to mention the procedure and related configuration in the docs, by labeling them as experimental.

Thanks.
Matteo

**Comment by swoehrl-mw** [comment]
Hi @mcaliandro. That is literally the intended usecase for the operator. You deploy it once per kubernetes cluster (for example in some central infrastructure or base namespace) and then use the CRDs to deploy and manage any number of opensearch clusters in all namespaces.

There is nothing extra you need to configure for this. Just deploy the operator, then start deploying CRs for opensearch clusters.

**Comment by kfox1111** [comment]
Can it be used only within a single namespace to manage clusters? We've found, sometime the olm model of install helps prevent problems when upgrading a dev os cluster in a dev namespace using a new version of the operator and not break a prod os cluster in a prod namespace with the previous version of the operator, all within the same k8s cluster.

**Reply by swoehrl-mw** [reply]
You can limit the operator to watch only CRs from one namespace by setting `manager.watchNamespace` in the operator helm chart values during installation. It is also possible to run several instances of the operator in parallel, but at the moment you need to take care to only install the CRDs once.

Running several versions of the operator at the same time will be problematic as with each new version we update and extend the CRDs, and these are cluster-level. So if you update only one of several running operator instances you'd have either old operator running with new CRDs or new operator running with old CRDs. That will likely produce problems. I strongly suggest to to separate dev/prod by clusters and not by namespaces.
